src/sdk/ConversationTranslator/Conversation.ts

```diff
@@ -38,124 +52,612 @@ export abstract class Conversation implements IConversation {
+        if (!profanity) {
+           speechConfig.setProfanity(ProfanityOption.Masked);
+        }
+
+        // check the nickname: it should pass this regex: ^\w+([\s-][\w\(\)]+)*$"
+        // TODO: specify the regex required. Nicknames must be unique or get the duplicate nickname error
+        // TODO: check what the max length is and if a truncation is required or if the service handles it without an error
+        let hostNickname: string = speechConfig.getProperty(PropertyId[PropertyId.ConversationTranslator_Name]);
+        if (hostNickname === undefined || hostNickname === null || hostNickname.length <= 1 || hostNickname.length > 50) {
+            hostNickname = "Host";
+        }
+        speechConfig.setProperty(PropertyId[PropertyId.ConversationTranslator_Name], hostNickname);
+
+        // save the speech config for future usage
+        this.privConfig = speechConfig;
+
+        // save the config properties
+        const configImpl = speechConfig as SpeechTranslationConfigImpl;
+        this.privProperties = configImpl.properties.clone();
+
     }
 
+    /***
+     * Create a new conversation as Host
+     */
+    public createConversationAsync(cb?: () => void, err?: (e: string) => void): void {
+        try {
+
+            this.reset();
+
+            this.privManager.createOrJoin(this.privProperties, undefined,
+                ((room: IInternalConversation) => {
+
+                    this.privRoom = room;
+
+                    if (!!cb) {
+                        try {
+                            cb();
+                        } catch (e) {
+                            if (!!err) {
+                                err(e);
+                            }
+                        }
+                        cb = undefined;
+                    }
+                }),
+                ((error: any) => {
+                    if (!!err) {
+                        if (error instanceof Error) {
+                            const typedError: Error = error as Error;
+                            err(typedError.name + ": " + typedError.message);
+
+                        } else {
+                            err(error);
+                        }
+                    }
+                }));
+
+        } catch (error) {
+            if (!!err) {
+                if (error instanceof Error) {
+                    const typedError: Error = error as Error;
+                    err(typedError.name + ": " + typedError.message);
+
+                } else {
+                    err(error);
+                }
+            }
+        }
+    }
+
+    /**
+     * Starts a new conversation as host.
+     * @param cb
+     * @param err
+     */
     public startConversationAsync(cb?: () => void, err?: (e: string) => void): void {
 
-        throw new Error("Method not implemented.");
+        try {
+
+            // connect
+            this.privConnection = new ConversationConnection(this.privConfig);
```

Comment:
Have you considered checking if this already set to protect against calling this method twice?

Reply:
I've added a check - need to add some additional Tests too.